pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -129,44 +129,43 @@ type NodeInfo struct {
 }
 
 // New creates an instance of a mon cluster
-func New(
-	clusterInfo *cephconfig.ClusterInfo,
-	context *clusterd.Context,
-	namespace, dataDirHostPath, rookVersion string,
-	cephVersion cephv1.CephVersionSpec,
-	mon cephv1.MonSpec,
-	placement rookalpha.Placement,
-	hostNetwork bool,
-	resources v1.ResourceRequirements,
-	ownerRef metav1.OwnerReference,
-) *Cluster {
+func New(context *clusterd.Context, namespace, dataDirHostPath string, hostNetwork bool, ownerRef metav1.OwnerReference) *Cluster {
 	return &Cluster{
-		clusterInfo:          clusterInfo,
-		context:              context,
-		placement:            placement,
-		dataDirHostPath:      dataDirHostPath,
-		Namespace:            namespace,
-		rookVersion:          rookVersion,
-		cephVersion:          cephVersion,
-		Count:                mon.Count,
-		AllowMultiplePerNode: mon.AllowMultiplePerNode,
-		maxMonID:             -1,
-		waitForStart:         true,
-		monPodRetryInterval:  6 * time.Second,
-		monPodTimeout:        5 * time.Minute,
-		monTimeoutList:       map[string]time.Time{},
-		HostNetwork:          hostNetwork,
+		context:             context,
+		dataDirHostPath:     dataDirHostPath,
+		Namespace:           namespace,
+		maxMonID:            -1,
+		waitForStart:        true,
+		monPodRetryInterval: 6 * time.Second,
+		monPodTimeout:       5 * time.Minute,
+		monTimeoutList:      map[string]time.Time{},
+		HostNetwork:         hostNetwork,
 		mapping: &Mapping{
 			Node: map[string]*NodeInfo{},
 			Port: map[string]int32{},
 		},
-		resources: resources,
-		ownerRef:  ownerRef,
+		ownerRef: ownerRef,
 	}
 }
 
 // Start begins the process of running a cluster of Ceph mons.
-func (c *Cluster) Start() (*cephconfig.ClusterInfo, error) {
+func (c *Cluster) Start(clusterInfo *cephconfig.ClusterInfo, rookVersion string, cephVersion cephv1.CephVersionSpec,
+	mon cephv1.MonSpec, placement rookalpha.Placement, resources v1.ResourceRequirements) (*cephconfig.ClusterInfo, error) {
+
+	// Only one goroutine can orchestrate the mons at a time
+	logger.Infof("Acquiring lock for mon orchestration")
+	c.monMutex.Lock()
+	defer c.monMutex.Unlock()
+	logger.Infof("Acquired lock for mon orchestration")
+
+	c.clusterInfo = clusterInfo
```

Comment:
Here, there is a lot of boilerplate to copy various cluster CRD items into the mon's `Cluster`. Is there a reason not to just use the cluster spec?

Reply:
Agreed, when i'm simplifying the params i'll look at reworking this.

Reply:
This is also used by `health.go`, perhaps a helper method makes sense? Later it could serve other purpose too.

Reply:
@leseb What helper would make sense here? The health.go doesn't update these properties, it just consumes them.

Reply:
@travisn I meant a helper method to acquire the lock.

Reply:
I think you are talking about something like this, Seb?

```go
func (c *Cluster) acquireOrchestrationLock() {
	logger.Debugf("Acquiring lock for mon orchestration")
	c.orchestrationMutex.Lock()
	logger.Debugf("Acquired lock for mon orchestration")
}

func (c *Cluster) releaseOrchestrationLock() {
	c.orchestrationMutex.Unlock()
	logger.Debugf("Released lock for mon orchestration")
}
```

and then the 4 lines above become:
```go
c.acquireOrchestrationLock()
defer c.releaseOrchestrationLock()
```

It at least makes the debug messaging standard for whatever method is doing the locking/unlocking.

Reply:
Exactly 👍